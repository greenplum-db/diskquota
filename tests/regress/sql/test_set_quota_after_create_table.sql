\! gpconfig -c shared_preload_libraries -v '' > /dev/null 
\! gpstop -far > /dev/null
\c

CREATE ROLE test SUPERUSER;

SET ROLE test;

-- Create table "t" with diskquota disabled
CREATE TABLE t (i) AS SELECT generate_series(1, 100000);

\! gpconfig -c shared_preload_libraries -v 'diskquota'> /dev/null 
\! gpstop -far > /dev/null
\c

SET ROLE test;

-- Init table_size to include table "t"
SELECT diskquota.init_table_size_table();

SELECT tableid::regclass, size FROM diskquota.table_size
WHERE tableid = 't'::regclass AND segid = -1;

SELECT 't'::regclass::oid;

-- Ensure that table "t" is not active
SELECT diskquota.diskquota_fetch_table_stat(0, ARRAY[]::oid[])
FROM gp_dist_random('gp_id');

SELECT diskquota.set_role_quota(current_role, '1MB');

SELECT diskquota.wait_for_worker_new_epoch();

-- Expect that current role is in the blackmap 
SELECT rolname FROM pg_authid, diskquota.blackmap WHERE oid = target_oid;

SELECT diskquota.set_role_quota(current_role, '-1');

SELECT diskquota.wait_for_worker_new_epoch();

SELECT rolname FROM pg_authid, diskquota.blackmap WHERE oid = target_oid;

DROP TABLE t;

RESET ROLE;

DROP ROLE test;
