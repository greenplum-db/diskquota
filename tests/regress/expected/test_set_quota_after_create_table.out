CREATE TABLE t (i) AS SELECT generate_series(1, 100000);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause. Creating a NULL policy entry.
SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

SELECT tableid::regclass, size FROM diskquota.table_size
WHERE tableid = 't'::regclass AND segid = -1;
 tableid |  size   
---------+---------
 t       | 3637248
(1 row)

-- Ensure that t is not active
SELECT diskquota.diskquota_fetch_table_stat(0, ARRAY[]::oid[])
FROM gp_dist_random('gp_id');
 diskquota_fetch_table_stat 
----------------------------
(0 rows)

SELECT diskquota.set_role_quota(current_role, '1MB');
 set_role_quota 
----------------
 
(1 row)

SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

SELECT rolname FROM pg_authid, diskquota.blackmap WHERE oid = target_oid;
 rolname 
---------
 gpadmin
(1 row)

DROP TABLE t;
SELECT diskquota.set_role_quota(current_role, '-1');
 set_role_quota 
----------------
 
(1 row)

SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

SELECT rolname FROM pg_authid, diskquota.blackmap WHERE oid = target_oid;
 rolname 
---------
(0 rows)

SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

